src/main/java/org/isf/lab/gui/LabEditExtended.java

```diff
@@ -639,29 +638,23 @@ private JButton getOkButton() {
 				} else if (examSelected.getProcedure() == 3) {
 					lab.setResult(examTextField.getText());
 				}
-				boolean result;
 				if (insert) {
 					lab.setAge(labPat.getAge());
 					try {
-						result = labManager.newLaboratory(lab, labRow);
+						labManager.newLaboratory(lab, labRow);
 					} catch (OHServiceException e1) {
 						OHServiceExceptionUtil.showMessages(e1);
-						return;
 					}
 				} else {
 					try {
-						result = labManager.updateLaboratory(lab, labRow);
+						labManager.updateLaboratory(lab, labRow);
+						fireLabUpdated();
+						dispose();
 					} catch (OHServiceException e1) {
+						MessageDialog.error(null, "angal.common.datacouldnotbesaved.msg");
```

Comment:
shouldn't work this way?

```suggestion
						MessageDialog.error(this, "angal.common.datacouldnotbesaved.msg");
```